sdks/go/pkg/beam/core/runtime/xlangx/expansionx/download_test.go

```diff
@@ -0,0 +1,131 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expansionx
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestGetURLForBeamJar(t *testing.T) {
+	tests := []struct {
+		name        string
+		target      string
+		version     string
+		expectedURL URL
+	}{
+		{
+			"base",
+			":sdks:java:fake",
+			"VERSION",
+			"https://repo.maven.apache.org/maven2/org/apache/beam/beam-sdks-java-fake/VERSION/beam-sdks-java-fake-VERSION.jar",
+		},
+		{
+			"versioned",
+			":sdks:java:fake",
+			"1.2.3",
+			"https://repo.maven.apache.org/maven2/org/apache/beam/beam-sdks-java-fake/1.2.3/beam-sdks-java-fake-1.2.3.jar",
+		},
+	}
+	for _, test := range tests {
+		madeURL, _ := getURLForBeamJar(test.target, test.version)
+		if madeURL != test.expectedURL {
+			t.Errorf("test %v failed: wanted URL %v, got %v", test.name, test.expectedURL, madeURL)
+		}
+	}
+}
+
+func TestDropEndOfGradleTarget(t *testing.T) {
+	target := ":sdks:java:fake:runFake"
+	expected := ":sdks:java:fake"
+	returned := dropEndOfGradleTarget(target)
+	if returned != expected {
+		t.Errorf("wanted %v, got %v", expected, returned)
+	}
+}
+
+func TestGetCacheDir(t *testing.T) {
+	cacheDir := getCacheDir()
+	if !strings.Contains(cacheDir, jarCache[2:]) {
+		t.Errorf("failed to get cache directory: wanted %v, got %v", jarCache[:2], cacheDir)
+	}
+}
+
+func TestCheckDir(t *testing.T) {
+	d, err := ioutil.TempDir(os.Getenv("TEST_TMPDIR"), "expansionx-*")
+	if err != nil {
+		t.Fatalf("failed to make temp directory, got %v", err)
+	}
+	defer os.RemoveAll(d)
```

Comment:
While in this instance, it should be identical, in tests, prefer using `t.Cleanup()`, since that allows the test framework to ensure the cleanup happens, rather than function exit. Here and the other test defers.

eg.
`t.Cleanup(func() { os.RemoveAll(d) })`

Reply:
Changed